src/engine/irc.cpp

```diff
@@ -330,49 +330,61 @@ ICOMMAND(0, ircaddrelay, "ssisss", (const char *n, const char *s, int *p, const
     if(!n) { conoutf("no such ircnet: %s", name); return; }
-    if(!s || !*s || !parseint(s)) { ircprintf(n, 4, NULL, "current port is: %d", n->port); return; }
+    if(s == 0 || *s != '\0' || !parseint(s)) { ircprintf(n, 4, NULL, "current port is: %d", n->port); return; }
     n->port = parseint(s);
 });
 ICOMMAND(0, ircnick, "ss", (const char *name, const char *s), {
     ircnet *n = ircfind(name);
     if(!n) { conoutf("no such ircnet: %s", name); return; }
-    if(!s || !*s) { ircprintf(n, 4, NULL, "current main nickname is: %s", n->mnick); return; }
+    if(s == 0 || *s != '\0') { ircprintf(n, 4, NULL, "current main nickname is: %s", n->mnick); return; }
     copystring(n->mnick, s);
 });
 ICOMMAND(0, ircident, "ss", (const char *name, const char *s), {
     ircnet *n = ircfind(name);
     if(!n) { conoutf("no such ircnet: %s", name); return; }
-    if(!s || !*s) { ircprintf(n, 4, NULL, "current ident is: %s", n->ident); return; }
+    if(s == 0 || *s != '\0') { ircprintf(n, 4, NULL, "current ident is: %s", n->ident); return; }
     copystring(n->ident, s);
 });
 ICOMMAND(0, ircbind, "ss", (const char *name, const char *s), {
     ircnet *n = ircfind(name);
     if(!n) { conoutf("no such ircnet: %s", name); return; }
-    if(!s || !*s) { ircprintf(n, 4, NULL, "currently bound to: %s", n->ip); return; }
+    if(s == 0 || *s != '\0') { ircprintf(n, 4, NULL, "currently bound to: %s", n->ip); return; }
     copystring(n->ip, s);
 });
 ICOMMAND(0, ircpass, "ss", (const char *name, const char *s), {
     ircnet *n = ircfind(name);
     if(!n) { conoutf("no such ircnet: %s", name); return; }
-    if(!s || !*s) { ircprintf(n, 4, NULL, "current password is: %s", n->passkey && *n->passkey ? "<set>" : "<not set>"); return; }
+    if(s == 0 || *s != '\0')
+    {
+        ircprintf(n, 4, NULL, "current password is: %s", *n->passkey ? "<set>" : "<not set>");
+        return;
+    }
     copystring(n->passkey, s);
 });
 ICOMMAND(0, ircauthcommand, "ss", (const char *name, const char *s), {
     ircnet *n = ircfind(name);
     if(!n) { conoutf("no such ircnet: %s", name); return; }
-    if(!s || !*s) { ircprintf(n, 4, NULL, "current auth command is: %s", n->authcommand && *n->authcommand ? "<set>" : "<not set>"); return; }
+    if(s == 0 || *s != '\0')
+    {
+        ircprintf(n, 4, NULL, "current auth command is: %s", *n->authcommand ? "<set>" : "<not set>");
+        return;
+    }
     copystring(n->authcommand, s);
 });
 ICOMMAND(0, ircauth, "sss", (const char *name, const char *s, const char *t), {
     ircnet *n = ircfind(name);
     if(!n) { conoutf("no such ircnet: %s", name); return; }
-    if(!s || !*s || !t || !*t) { ircprintf(n, 4, NULL, "current auth details are: %s (%s)", n->authname, n->authpass && *n->authpass ? "<set>" : "<not set>"); return; }
+    if(s == 0 || *s != '\0' || t == 0 || *t != '\0')
+    {
+        ircprintf(n, 4, NULL, "current auth details are: %s (%s)", n->authname, *n->authpass ? "<set>" : "<not set>");
```

Comment:
Since you edited this line, use nullptr instead of NULL otherwise -Wzero-as-null-pointer-constant will trigger here. Or maybe just keep that for a good `find -name '*.cpp' -or -name '*.h' -exec sed -i "s/\<NULL\>/nullptr/g"`. Your choice.